.github/workflows/platformDeploy.yml

```diff
@@ -0,0 +1,270 @@
+          sudo python3 -m pip install --upgrade pip==20.3.4
+          pip3 install wheel # need wheel before cloudflare, this is the only way to ensure order.
+          pip3 install cloudflare
+
+      - name: Configure AWS Credentials
+        # Version: 1.5.5
+        uses: aws-actions/configure-aws-credentials@e97d7fbc8e0e5af69631c13daa0f4b5a8d88165b
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: us-east-1
+
+      - name: Install node packages
+        uses: nick-invision/retry@7c68161adf97a48beb850a595b8784ec57a98cbb
+        with:
+          timeout_minutes: 10
+          max_attempts: 5
+          command: npm ci
+
+      - name: Build web for production
+        if: ${{ env.SHOULD_DEPLOY_PRODUCTION == 'true' }}
+        run: npm run build
+
+      - name: Deploy production to S3
+        if: ${{ env.SHOULD_DEPLOY_PRODUCTION == 'true' }}
+        run: aws s3 cp --recursive --acl public-read $GITHUB_WORKSPACE/dist s3://expensify-cash/
+
+      - name: Build web for staging
+        if: ${{ env.SHOULD_DEPLOY_PRODUCTION == 'false' }}
+        run: npm run build-staging
+
+      - name: Deploy staging to S3
+        if: ${{ env.SHOULD_DEPLOY_PRODUCTION == 'false' }}
+        run: aws s3 cp --recursive --acl public-read $GITHUB_WORKSPACE/dist s3://staging-expensify-cash/
+
+      - name: Purge production Cloudflare cache
+        if: ${{ env.SHOULD_DEPLOY_PRODUCTION == 'true' }}
+        run: /home/runner/.local/bin/cli4 --delete hosts=["expensify.cash"] /zones/:357b42c0989e3bed82f6d6400e8b9d66/purge_cache
+        env:
+          CF_API_KEY: ${{ secrets.CLOUDFLARE_TOKEN }}
+
+      - name: Purge staging Cloudflare cache
+        if: ${{ env.SHOULD_DEPLOY_PRODUCTION == 'false' }}
+        run: /home/runner/.local/bin/cli4 --delete hosts=["staging.expensify.cash"] /zones/:357b42c0989e3bed82f6d6400e8b9d66/purge_cache
+        env:
+          CF_API_KEY: ${{ secrets.CLOUDFLARE_TOKEN }}
+
+  postSlackMessage:
+    name: Post a Slack message when any platform fails to build or deploy
+    runs-on: ubuntu-latest
+    if: ${{ failure() }}
+    needs: [android, desktop, iOS, web]
+    steps:
+      - uses: 8398a7/action-slack@v3
+        name: Job failed Slack notification
+        with:
+          status: custom
+          fields: workflow, repo
+          custom_payload: |
+            {
+              channel: '#announce',
+              attachments: [{
+                color: "#DB4545",
+                pretext: `<!here>`,
+                text: `💥 ${process.env.AS_REPO} failed on ${process.env.AS_WORKFLOW} workflow 💥`,
+              }]
+            }
+        env:
+          GITHUB_TOKEN: ${{ github.token }}
+          SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK }}
+
+  postGithubComment:
+    name: Post a GitHub comment when platforms are done building and deploying
```

Comment:
Note to @bondydaa that right now this job just logs a message, doesn't actually post anything in GitHub. I think @AndrewGable is planning on doing that in a follow-up.

Reply:
Correct, this is just a first step PR.